tests/virt_autotest/setup_dns_service.pm

```diff
@@ -0,0 +1,191 @@
+    my %vm_hash_forward             = "";
+    my %vm_hash_reverse             = "";
+    my $get_vm_hostnames_inactive   = "virsh list --inactive | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames_inactive       = script_output($get_vm_hostnames_inactive, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_inactive_array = split(/\n+/, $vm_hostnames_inactive);
+    my $get_vm_hostnames            = "virsh list  --all | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames                = script_output($get_vm_hostnames, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_array          = split(/\n+/, $vm_hostnames);
+    my $get_vm_macaddress           = "";
+    my @vm_macaddresses_array       = "";
+    my $get_vm_ipaddress            = "";
+    my $vm_ipaddress                = "";
+    my $vm_ipaddress_lastpart       = "";
+    my $vm_hash_index               = "";
+    foreach (@vm_hostnames_array)
+    {
+        if ($_ ~~ @vm_hostnames_inactive_array) {
+            assert_script_run("virsh start $_");
+        }
+        else {
+            assert_script_run("virsh reboot $_");
+        }
+    }
+    wait_still_screen 10;
+    foreach (@vm_hostnames_array)
+    {
+        $get_vm_macaddress = "virsh domiflist --domain $_ | grep -oE \"([0-9|a-z]{2}:){5}[0-9|a-z]{2}\"";
+        $vm_macaddresses_array[$vm_hash_index] = script_output($get_vm_macaddress, $wait_script, type_command => 0, proceed_on_failure => 0);
+        $get_vm_ipaddress = "cat $dhcpd_lease_file | awk \'!(\$0 in S) {print; S[\$0]}\' | grep -iE \"$vm_macaddresses_array[$vm_hash_index]\" -B8 | grep -oE \"([0-9]{1,3}\\.){1,3}[0-9]{1,3}\"";
+        $vm_ipaddress = script_output($get_vm_ipaddress, $wait_script, type_command => 0, proceed_on_failure => 0);
+        ($vm_ipaddress_lastpart) = $vm_ipaddress =~ /\d+\.\d+\.\d+\.(\d+)/;
+        $vm_hash_forward{$_} = $vm_ipaddress;
+        $vm_hash_reverse{$_} = $vm_ipaddress_lastpart;
+        $vm_hash_index += 1;
+    }
+    delete @vm_hash_forward{grep { not defined $vm_hash_forward{$_} } keys %vm_hash_forward};
+    delete @vm_hash_reverse{grep { not defined $vm_hash_reverse{$_} } keys %vm_hash_reverse};
+
+    my $dns_forward_zone_file = "/var/lib/named/$dns_domain_forward.zone";
+    my $dns_reverse_zone_file = "/var/lib/named/$dns_domain_reverse.zone";
+    my $dns_record            = "";
+    my $dns_records_forward   = "";
+    my $dns_records_reverse   = "";
+
+    foreach (keys %vm_hash_forward)
+    {
+        $dns_record = join("\\\\t", $_, "IN", "A", $vm_hash_forward{$_});
+        $dns_records_forward = join("\\\\n", $dns_records_forward, $dns_record);
+    }
+
+    #The content of testvirt.net zone file
+    my $dns_forward_zone = "\; Authoritative data for $dns_domain_forward zone
+\\\$ORIGIN         $dns_domain_forward.
+\\\$TTL 1D
+\@   IN SOA  ns.$dns_domain_forward.  mailmaster.$dns_domain_forward. (
```

Comment:
That day, in our talk, I mentioned mail part in zone file, i confirmed with you that it was not necessary. And we agreed to remove it as first step to decrease time. I thought it was this part, mailmaster. Was it?

Reply:
This does not make sense. I already removed the part that is not necessary.

Reply:
In case you did not remember what the previous one looks like:
    my $dns_reverse_zone = "\; Authoritative data for $dns_domain_reverse.in-addr.arpa zone
\; ?
\\\$ORIGIN $dns_domain_reverse.in-addr.arpa.
\\\$TTL 1D
\@          IN SOA ns.$dns_domain_forward. mailmaster.$dns_domain_forward. (
              \$(date \\+\\\%Y\\\%m\\\%d)        ; serial
              21600             ; refresh
              3600              ; retry
              3600000           ; expire
              86400 )           ; minimum


\; ----------- ENREGISTREMENTS -----------
\@                       IN NS                   ns.$dns_domain_forward.
\;\;
\$(echo -e $dns_records_reverse)
\;\;
\; ----------- ENREGISTREMENTS SPECIAUX -----------";

Reply:
I will try also remove the "mailmaster.$dns_domain_forward." in line 89.